### PR TITLE
implement missing_values_found rule

### LINKED ISSUE #39: New validation rule, missing_values_found
Specification: https://github.com/Big-Life-Lab/PHES-ODM-Validation/blob/main/validation-rules/missing_values_found.md

### PR DESCRIPTION
**Pull request by zargot**
fixes #39 

- rename custom cerb rule trimEmpty to emptyTrimmed
- change emptyTrimmed value to False to match existing empty rule semantics
- add functionality for reporting warnings from rules
- add functionality for matching multiple keys to one rule
- add "gen_conditional_schema" helper

### CODE REVIEW

File: src/odm_validation/rules.py [hunk 1]
```diff
@@ -45,12 +45,15 @@ class Rule:
     - value_type
     """
     id: str
-    key: str
+    keys: List[str]
+    is_warning: bool
+    match_all_keys: bool
```
Comment (yulric) [inline src/odm_validation/rules.py:50]: I'm confused about why we need the match_all_keys field in the Rule object? missing_values_found is the only one that actually uses that but its also the only one whose keys field has multiple values.

**Review by yulric — CHANGES_REQUESTED** [review]
1. It looks like you're missing the `invalidValue` field in the `error-report-1.json` file
2. Can you update the specification file with the rename from trimEmpty to emptyTrimmed

File: src/odm_validation/rules.py (hunk 1 above)
Comment (zargot) [reply src/odm_validation/rules.py:50]: the rules with type/coerce also have multiple keys

File: src/odm_validation/part_tables.py [hunk 2]
```diff
@@ -267,27 +285,40 @@ def get_table_id(part: dict) -> Optional[str]:
...
 
-def filter_compatible(parts: Dataset, version: Version) -> Dataset:
+def filter_compatible(parts: Dataset, schema_version: Version) -> Dataset:
     """Filters `parts` by `version`."""
     result = []
     for row in parts:
         part_id = get_partID(row)
-        if not (is_compatible(row, version)):
+        first, last = get_version_range(row)
+        active: bool = row.get('status') == 'active'
```
[the remark this answers is not in the document]
Comment (zargot) [reply src/odm_validation/part_tables.py]: done

File: src/odm_validation/part_tables.py [hunk 3]
```diff
@@ -131,23 +134,33 @@ def parse_row_version(row, field, default=None):
...
-    v = version
+    v = schema_version
     if v.compare(first) < 0:
         return False
     if v.compare(last) < 0:
         return True
     return active
 
 
+def part_is_compatible(part: Part, version: Version) -> bool:
```
[the remark this answers is not in the document]
Comment (zargot) [reply src/odm_validation/part_tables.py]: done

**Review by yulric — APPROVED** [review]

**Pull request merged**